solutions/src/main/java/ru/sberbank/school/task08/state/GameObject.java

```diff
@@ -0,0 +1,55 @@
+package ru.sberbank.school.task08.state;
+
+import java.util.Objects;
+
+public class GameObject implements InstantiatableEntity {
+    private final Type type;
+    private final Status status;
+    private long hitPoints;
+
+    public GameObject(Type type, Status status, long hitPoints) {
+        this.type = type;
+        this.status = status;
+        this.hitPoints = hitPoints;
+    }
+
+    @Override
+    public InstantiatableEntity getInstance(Type type, Status status, long hitPoints) {
+        return new GameObject(type, status, hitPoints);
+    }
+
+    public Type getType() {
+        return this.type;
+    }
+
+    public Status getStatus() {
+        return this.status;
+    }
+
+    public long getHitPoints() {
+        return hitPoints;
+    }
+
+    public void setHitPoints(long hitPoints) {
+        this.hitPoints = hitPoints;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
&#39;if&#39; construct must use &#39;{}&#39;s.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a56173481>*